test/parallel_api/algorithm/alg.modifying.operations/shift_left_right.pass.cpp

```diff
@@ -0,0 +1,182 @@
+    }
+#endif
+};
+
+template <typename T, typename Size, typename Algo>
+void
+test_shift_by_type(Size m, Size n, Algo algo)
+{
+    TestUtils::Sequence<T> orig(m, [](::std::size_t v) -> T { return T(v); }); //fill data
+    TestUtils::Sequence<T> in(m, [](::std::size_t v) -> T { return T(v); }); //fill data
+
+    TestUtils::invoke_on_all_policies<>()(test_shift<T>(), in.begin(), m, orig.begin(), n, algo);
+}
+
+struct shift_left_algo
+{
+    template <typename Policy, typename It, typename Size>
+    It operator()(Policy&& exec, It first, It last, Size n)
+    {
+        return oneapi::dpl::shift_left(::std::forward<Policy>(exec), first, last, n);
+    }
+
+    template <typename It, typename Size>
+    void
+    check(It res, It first, Size m, It first_exp, Size n)
+    {
+        //if (n > 0 && n < m), returns first + (m - n). Otherwise, if n  > 0, returns first.
+        //Otherwise, returns last.
+        It __last = std::next(first, m);
+        auto res_exp = (n > 0 && n < m ? std::next(first, m - n) : (n > 0 ? first : __last));
+
+        EXPECT_TRUE(res_exp == res, "wrong return value of shift_left");
+
+        if(res != first && res != __last)
+        {
+            EXPECT_EQ_N(first, std::next(first_exp, + n), m - n, "wrong effect of shift_left");
+            //restore unput data
+            std::copy_n(first_exp, m, first);
+        }
+    }
+};
+
+struct shift_right_algo
+{
+    template <typename Policy, typename It, typename Size>
+    typename ::std::enable_if<TestUtils::is_same_iterator_category<It, ::std::bidirectional_iterator_tag>::value
+                            || TestUtils::is_same_iterator_category<It, ::std::random_access_iterator_tag>::value,
+                            It>::type
+    operator()(Policy&& exec, It first, It last, Size n)
+    {
+        return oneapi::dpl::shift_right(::std::forward<Policy>(exec), first, last, n);
+    }
+    //skip the test for non-bidirectional iterator (forward iterator, etc)
+    template <typename Policy, typename It, typename Size>
+    typename ::std::enable_if<!TestUtils::is_same_iterator_category<It, ::std::bidirectional_iterator_tag>::value
+                            && !TestUtils::is_same_iterator_category<It, ::std::random_access_iterator_tag>::value,
+                            It>::type
+    operator()(Policy&& exec, It first, It last, Size n)
+    {
+        return first;
+    }
+
+    template <typename It, typename Size>
+    typename ::std::enable_if<TestUtils::is_same_iterator_category<It, ::std::bidirectional_iterator_tag>::value
+                            || TestUtils::is_same_iterator_category<It, ::std::random_access_iterator_tag>::value,
+                            void>::type
+    check(It res, It first, Size m, It first_exp, Size n)
+    {
+        //if (n > 0 && n < m), returns first + n. Otherwise, if n  > 0, returns last.
+        //Otherwise, returns firts.
+        It __last = std::next(first, m);
```

Comment:
What do we do with `std` in tests? should it be `::std`

Reply:
fixed